### ISSUE TITLE
Request Headers Duplicating.

### ISSUE THREAD

**Issue by shaggi7** [processone/ejabberd#3056]
I have build a HTTP module in ejabberd. 
I have mentioned the headers in the ejabberd.yml file. 
I have tried two configurations:
Config 1:
```
listen:
  -
    port: 5222
    ip: "::"
    module: ejabberd_c2s
    max_stanza_size: 262144
    shaper: c2s_shaper
    access: c2s
    starttls_required: true
  -
    port: 5269
    ip: "::"
    module: ejabberd_s2s_in
    max_stanza_size: 524288
  -
    port: 5443
    ip: "::"
    module: ejabberd_http
    tls: true
    request_handlers:
      /admin: ejabberd_web_admin
      /api: mod_http_api
      /bosh: mod_bosh
      /captcha: ejabberd_captcha
      /upload: mod_http_upload
      /ws: ejabberd_http_ws
      /oauth: ejabberd_oauth
    custom_headers:
      "Access-Control-Allow-Origin": "*"
      "Access-Control-Allow-Methods": "OPTIONS, HEAD, GET, PUT"
      "Access-Control-Allow-Headers": "Authorization, Content-Type"
      "Access-Control-Allow-Credentials": "true"
  -
    port: 5280
    ip: "::"
    module: ejabberd_http
    request_handlers:
        /admin: ejabberd_web_admin
        /.well-known/acme-challenge: ejabberd_acme
        /http-bind: mod_bosh
        /upload: mod_http_upload
        /users: mod_hello_world
     custom_headers:
        "Access-Control-Allow-Origin": "*"
        "Access-Control-Allow-Methods": "OPTIONS, HEAD, GET, PUT"
        "Access-Control-Allow-Headers": "Authorization, Content-Type"
        "Access-Control-Allow-Credentials": "true"
```
Config 2:
```
listen:
  -
    port: 5222
    ip: "::"
    module: ejabberd_c2s
    max_stanza_size: 262144
    shaper: c2s_shaper
    access: c2s
    starttls_required: true
  -
    port: 5269
    ip: "::"
    module: ejabberd_s2s_in
    max_stanza_size: 524288
  -
    port: 5443
    ip: "::"
    module: ejabberd_http
    tls: true
    request_handlers:
      /admin: ejabberd_web_admin
      /api: mod_http_api
      /bosh: mod_bosh
      /captcha: ejabberd_captcha
      /upload: mod_http_upload
      /ws: ejabberd_http_ws
      /oauth: ejabberd_oauth
    custom_headers:
      "Access-Control-Allow-Origin": "*"
      "Access-Control-Allow-Methods": "OPTIONS, HEAD, GET, PUT"
      "Access-Control-Allow-Headers": "Authorization, Content-Type"
      "Access-Control-Allow-Credentials": "true"
  -
    port: 5280
    ip: "::"
    module: ejabberd_http
    request_handlers:
        /admin: ejabberd_web_admin
        /.well-known/acme-challenge: ejabberd_acme
        /http-bind: mod_bosh
        /upload: mod_http_upload
        /users: mod_hello_world
     custom_headers:
        #"Access-Control-Allow-Origin": "*"
        "Access-Control-Allow-Methods": "OPTIONS, HEAD, GET, PUT"
        "Access-Control-Allow-Headers": "Authorization, Content-Type"
        "Access-Control-Allow-Credentials": "true
```
Only the line "Access-Control-Allow-Origin" under port 5280 is commented in the Config 2.

I am using strophe which uses 'http-bind' for the connection in keepalive state.
The thing is that Config 1 is creating problem with login as it say 
```
Access to XMLHttpRequest at 'http://192.168.4.69:5280/http-bind' from origin 'http://localhost:3000' has been blocked by CORS policy: The 'Access-Control-Allow-Origin' header contains multiple values '*, *', but only one is allowed.
```
But make http request successfully
And 
Config 2 is making login succesfully but giving error in API request as follows
```
Access to fetch at 'http://192.168.4.69:5280/users/registered-users' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

What could be the error.

**Comment by prefiks** [comment]
I have branch with this feature mostly implemented, i will finish it before next release

**Comment by shaggi7** [comment]
Ok Thanks. Can you comment on this on the release date so i can get notified???

**Comment by prefiks** [comment]
This should be fixed by 7eb35d93e5640d7e85378a9f8bed68c6cbf966af
